### PR TITLE
Botfixes: Cleans up the worst bot in the game, Floorbots

### PR DESCRIPTION
**Pull request by Tsar-Salat**
Ports:
- https://github.com/tgstation/tgstation/pull/52772
- https://github.com/tgstation/tgstation/pull/54786
- https://github.com/tgstation/tgstation/pull/54885

## About The Pull Request

This is a general fix pr for floorbots. Its likely the last PR before I can TGUI all bots in general. Its just a bunch of ports that deal with floorbots.

### PR One
- No more negative tiles in floorbots inventory
- Turning off Replace Tiles will always stop it

### PR Two
- Fixes floorbot pathfinding. They wont build ontop of another floorbot anymore, ever. If they ever conflict they will whistle at eachother before moving away.

### PR Three 
- Floorbots wont apply tiles to already tiled turfs (unless custom tiles obv)
- Fixes floorbots spewing lattice when deconstructing floor into plating
- Completely overhauled the floorbots magnet system. Now they wont weirdly anchor to the floor, but go through a proc entirely. The achored usages have mostly been replaced with `move_resist` and `TRAIT_IMMOBILIZE`. Should fix them getting stuck as well

### Misc Changes
- Converted them to use update_appearance
- changes a spawn to a timer

## Why It's Good For The Game
These bots suck. This should fix some of their buggiest behaviors, and make them much more reliable.

Good lead-in to TGUI'ing them and JPS improvements

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Fixing broken tiles

https://github.com/BeeStation/BeeStation-Hornet/assets/62388554/f5b41e82-aa8d-4925-a6ee-737c9ec971e4

Replacing with custom tiles

https://github.com/BeeStation/BeeStation-Hornet/assets/62388554/92cbb605-abda-4cb3-9ecd-0fed252896a9

</details>

## Changelog
:cl: rkz, Shizcalev, ArcaneMusic, Ghommie
fix: Fixed a scenario which could cause floorbots to end up with negative tiles in their inventory.
fix: Fixed floorbots not stopping the process of replacing tiles after turning off replace tiles while they're in the middle of replacing
fix: Floorbots had had a software update, preventing them from dogpiling on their target as easily as they did before.
soundadd: Floorbots will now play a small chime when stacked on top of each other to indicate that they're moving apart.
fix: Fixed floorbots applying floor tiles onto already tiled turfs.
fix: Fixed floorbot issues with material tiles.
fix: Fixed emagged floorbots spawning lattice when deconstructing floor into plating.
refactor: made floorbot magnets much more reliable
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/bot/floorbot.dm [hunk 1]
```diff
@@ -34,41 +32,50 @@
...
-	#define FIX_TILE		3
-	#define AUTO_TILE		4
-	#define PLACE_TILE		5
-	#define REPLACE_TILE		6
-	#define TILE_EMAG		7
+	#define HULL_BREACH	1
+	#define LINE_SPACE_MODE 2
+	#define FIX_TILE 3
+	#define AUTO_TILE 4
+	#define PLACE_TILE 5
```
Comment (itsmeow) [inline code/modules/mob/living/simple_animal/bot/floorbot.dm]: these need undefs

File: code/modules/mob/living/simple_animal/bot/floorbot.dm [hunk 2]
```diff
@@ -228,22 +245,22 @@
 		if(loc == target || loc == get_turf(target))
 			if(check_bot(target))	//Target is not defined at the parent
 				shuffle = TRUE
-				if(prob(50))	//50% chance to still try to repair so we dont end up with 2 floorbots failing to fix the last breach
+				if(prob(50)) //50% chance to still try to repair so we dont end up with 2 floorbots failing to fix the last breach
```
Comment (itsmeow) [inline code/modules/mob/living/simple_animal/bot/floorbot.dm:250]: Why not just check if the other bot `mode` is REPAIRING (check_bot_working)?

File: code/modules/mob/living/simple_animal/bot/floorbot.dm (hunk 1 above)
Comment (Tsar-Salat) [reply code/modules/mob/living/simple_animal/bot/floorbot.dm]: addressed in e5031b7

File: code/modules/mob/living/simple_animal/bot/floorbot.dm (hunk 2 above)
Comment (Tsar-Salat) [reply code/modules/mob/living/simple_animal/bot/floorbot.dm:250]: Idk, that not part of my changes

**Review by Rukofamicom — APPROVED** [review]
Seems good, just needs conflict resolved

(Likely conflicts with the anchored PR I just merged)

**Pull request merged**